src/Type/WPInterfaceTrait.php

```diff
@@ -40,6 +36,19 @@ protected function get_implemented_interfaces() {
 		 */
 		$interfaces = apply_filters( 'graphql_type_interfaces', $interfaces, $this->config, $this );
 
+		if ( empty( $interfaces ) || ! is_array( $interfaces ) ) {
+			return $interfaces;
+		}
+
+		// check if an interface is attempting to implement itself, and if so unset it
+		$key = array_search( strtolower( $this->config['name'] ), array_map( 'strtolower', $interfaces ), true );
+		if ( false !== $key ) {
+			graphql_debug( sprintf( __( 'The "%s" Interface attempted to implement itself, which is not allowed', 'wp-graphql' ), $interfaces[ $key ] ) );
+			unset( $interfaces[ $key ] );
+		}
+
+		$new_interfaces = [];
+
```

Comment:
(Moving https://github.com/wp-graphql/wp-graphql/commit/ca9f50f5a4765e2f7b8ee11280e74cdc80193213#r95832065 here for better discoverability)

> If I'm not mistaken, `in_array()` wouldn't work here because of the case sensitivity.
> 
> For better or worse you could implement an interface like:
> 
> `$interfaces = [ 'node', 'NODE', 'Node' ];`
> 
> And `unset( $interfaces[ 'Node' ] )` wouldn't unset `NODE`.
> 
> So the array search ensures we get the key for the interface regardless of case sensitivity.

It's precisely the `[ 'node', 'Node', 'NODE' ]` situation that is why I suggested suggesting ~`in_array()`~, since `array_search()` will only return the first match. We would still normalize the values.

edit: as of writing realized we dont need `in_array()` at all, and just rely on the existing foreach loop.
In untested psuedocode:

```php
$interfaces = apply_filters( 'graphql_type_interfaces', $interfaces, $this->config, $this );
// is_empty() on an array is handled by foreach() so we dont need it, but a different type returned is a coding error.
if ( ! is_array( $interfaces ) {
  graphql_debug( 'doing it wrong.....' );
  return []; // returning $interface if its something else breaks the method signature and would still throw a PHP error.
}

$normalized_interface_name = strtolower( $this->config['name'] );

foreach ( $interfaces as $interface ) {
  if ( ! is_string( $interface ) {
    continue;
  }

  // now lets test for dupes
  if ( in_array( strtolower( $interface), $normalized_interface_key, true ) {
    graphql_debug( 'Recursive interfaces are bad or whatever' );
    continue;
  }
  // rest of loop
}
```